src/View/Requirements_Backend.php

```diff
@@ -493,10 +501,50 @@ protected function getAllJavascript()
      */
     public function customScript($script, $uniquenessID = null)
     {
+        if ($uniquenessID) {
+            $this->customScript[$uniquenessID]['content'] = $script;
+        } else {
+            $this->customScript[] = $script;
+        }
+    }
+
+    /**
+     * Register the given Javascript code into the list of requirements with optional tag
+     * attributes.
+     *
+     * @param string $script The script content as a string (without enclosing `<script>` tag)
+     * @param array $options List of options. Available options include:
+     * - 'type' : Specifies the type of script
+     * - 'crossorigin' : Cross-origin policy for the resource
+     * @param string|int $uniquenessID A unique ID that ensures a piece of code is only added once
+     */
+    public function customScriptWithAttributes(string $script, array $attributes = [], string|int|null $uniquenessID = null)
+    {
+        // Get type
+        $type = null;
+        if (isset($this->customScriptAttributes[$uniquenessID]['type'])) {
+            $type = $this->customScriptAttributes[$uniquenessID]['type'];
+        }
+        if (isset($attributes['type'])) {
+            $type = strtolower($attributes['type']);
+        }
+
+        $crossorigin = $attributes['crossorigin'];
+        $crossorigin = isset($crossorigin) ? strtolower($crossorigin) : null;
+
         if ($uniquenessID) {
             $this->customScript[$uniquenessID] = $script;
+            $this->customScriptAttributess[$uniquenessID] = [
+                'type' => $type,
+                'crossorigin' => $crossorigin
+            ];
         } else {
             $this->customScript[] = $script;
+            $index = count($this->customScript) - 1;
+            $this->customScriptAttributes[$index] = [
+                'type' => $type,
+                'crossorigin' => $crossorigin
+            ];
```

Comment:
```suggestion
        $attrs = [];
        foreach (['type', 'crossorigin'] as $attrKey) {
            if (isset($attributes[$attrKey])) {
                $attrs[$attrKey] = strtolower($attributes[$attrKey]);
            }
        }
        if ($uniquenessID) {
            $this->customScript[$uniquenessID] = $script;
            $this->customScriptAttributes[$uniquenessID] = $attrs;
        } else {
            $this->customScript[] = $script;
            $index = count($this->customScript) - 1;
            $this->customScriptAttributes[$index] = $attrs;
        }
```
Fix the issue where `crossorigin` is a required key in $attributes when it shouldn't be